scipy/weave/c_spec.py

```diff
@@ -301,36 +300,30 @@ def c_to_py_code(self):
 #----------------------------------------------------------------------------
 # Standard Python numeric --> C type maps
 #----------------------------------------------------------------------------
-num_to_c_types = {}
-num_to_c_types[type(1)] = 'long'
-num_to_c_types[type(1.)] = 'double'
-num_to_c_types[type(1.+1.j)] = 'std::complex<double> '
+num_to_c_types = {type(1): 'long',
+                  type(1.): 'double',
+                  type(1. + 1.j): 'std::complex<double> ',
+                  long: 'npy_longlong',
+                  'T': 'T',
+                  'G': 'std::complex<longdouble> ',
+                  'F': 'std::complex<float> ',
+                  'D': 'std::complex<double> ',
+                  'g': 'npy_longdouble',
+                  'f': 'float',
+                  'd': 'double',
+                  'b': 'char', 'B': 'npy_ubyte',
+                  'h': 'short', 'H': 'npy_ushort',
+                  'i': 'int', 'I': 'npy_uint',
+                  '?': 'bool',
+                  'l': 'long', 'L': 'npy_ulong',
+                  'q': 'npy_longlong', 'Q': 'npy_ulonglong'}
+
+
 # !! hmmm. The following is likely unsafe...
-num_to_c_types[long] = 'npy_longlong'
```

Comment:
Here you lost the link between this line and the comment above. Can you move the comment up and explain that that's about ``long --> npy_longlong``?